addons/account/models/account_move.py

```diff
@@ -450,6 +450,7 @@ def _onchange_partner_id(self):
         if self.is_sale_document(include_receipts=True) and self.partner_id:
             self.invoice_payment_term_id = self.partner_id.property_payment_term_id or self.invoice_payment_term_id
             new_term_account = self.partner_id.commercial_partner_id.property_account_receivable_id
+            self.narration = self.company_id.with_context(lang=self.partner_id.lang).invoice_terms
```

Comment:
` or self.env.lang` to stay consistent ?

Reply:
I just added `or self.env.lang`, it should be good now :slightly_smiling_face: